File: src/main/com/etc/Person.java
package main.com.etc;

import org.jetbrains.annotations.NotNull;

public class Person {
    private String name;
    private int age;

    public Person (String name) {
        this.name = name;
    }

    public Person (String name, int age) {
        this.name = name;
        this.age = age;
    }

    @NotNull
    public String getName() {
        return name;
    }

    @NotNull
    public int getAge() {
        return age;
    }
}

File: src/main/com/etc/kotlin/Person.kt
package main.com.etc.kotlin

class Person(name: String, var age: Int) {

    /*val name: String = name
        get() = field.uppercase()*/

    var name = name
        set(value) {
            field = value.uppercase()
        }


    init {
        if (age <= 0) {
            throw IllegalArgumentException("나이는 ${age}일 수 없습니다.")
        }
    }

    constructor(name: String): this(name, 1) {
        println("첫번재 부생성자")
    }

    val isAdult: Boolean
        get() = this.age >= 20
}